### PR TITLE
chore(emotion,ui-i18n): throw warning when 'auto' dir (INSTUI-3306)

### PR DESCRIPTION
**Pull request by HerrTopi**
throw warning when 'auto' dir is passed to the TextDirectionContext

Closes: INSTUI-3306

### CODE REVIEW

**Review by matyasf — CHANGES_REQUESTED** [review]
BidirectionalProps still has a comment about 'auto' direction; ApplyTextDirectionProps still accepts 'auto'

**Review by matyasf — APPROVED** [review]

**Review by ke1k0 — CHANGES_REQUESTED** [review]
In the `ApplyTextDirection.test.tsx` it throws an error for the "auto" dir, please add a `ts-expect-error` that we intentionally pass it. It's also weird that this test passes, so we DO handle the auto value :)

File: packages/ui-i18n/src/ApplyTextDirection/__tests__/ApplyTextDirection.test.tsx [hunk 1]
```diff
@@ -92,19 +92,6 @@ describe('<ApplyTextDirection />', async () => {
...
-        <span>
-          هذه الفقرة باللغة العربية ، لذا يجب الانتقال من اليمين إلى اليسار.
-        </span>
-      </ApplyTextDirection>
-    )
-    expect(
-      getComputedStyle(subject.getDOMNode().childNodes[0] as Element).direction
-    ).to.equal('rtl')
-  })
-
```
Comment (matyasf) [inline packages/ui-i18n/src/ApplyTextDirection/__tests__/ApplyTextDirection.test.tsx:107]: if getTextDirection can return 'auto' we could move this test there

**Review by ke1k0 — APPROVED** [review]

**Review by matyasf — APPROVED** [review]
looks good, let's release

**Pull request merged**